src/Build/Resources/Strings.resx

```diff
@@ -264,11 +264,16 @@
     <comment>{StrBegin="MSB4114: "}UE: This message appears if the project file contains unreasonably nested Choose elements.
     LOCALIZATION: Do not localize "Choose" as it is an XML element name.</comment>
   </data>
-  <data name="CircularDependency" xml:space="preserve">
+  <data name="CircularDependencyInTargetGraph" xml:space="preserve">
     <value>MSB4006: There is a circular dependency in the target dependency graph involving target "{0}".</value>
     <comment>{StrBegin="MSB4006: "}UE: This message is shown when the build engine detects a target referenced in a circular manner -- a project cannot
     request a target to build itself (perhaps via a chain of other targets).</comment>
   </data>
+  <data name="CircularDependencyInTargetGraphWithVerbosity" xml:space="preserve">
+    <value>MSB4006: There is a circular dependency in the target dependency graph involving target "{0}". Target "{1}" has a "{2}" dependency on it, but it is depended upon by {3}.</value>
```

Comment:
If we end up with a different error, we'd need a new error code.

Reply:
My understanding is that this is the same error: Target Circular dependence. The only one reason why I add a new error message format is that the output is totally different from the old one. And I have no good idea to merge the different message output format into one. Any suggestions here?

Reply:
One idea is that if we can print the dependence graph when circular dependence failer is triggered. let me do more investigation and I will get back later.

Reply:
I think we don't need to add a new error code because this is a circular error of targets. I modify the key of the error message to improve the readability and merge the message format.

Reply:
I'm convinced.